src/main/java/io/cdap/plugin/gcp/gcs/ServiceAccountAccessTokenProvider.java

```diff
@@ -64,6 +120,10 @@ public void refresh() throws IOException {
     try {
       getCredentials().refresh();
     } catch (IOException e) {
+      if (isServerError(e)) {
+        throw new ServerErrorException(HttpStatus.SC_SERVICE_UNAVAILABLE, "Server error during refresh: " +
+          e.getMessage(), e);
+      }
       throw ErrorUtils.getProgramFailureException(new ErrorCategory(ErrorCategoryEnum.PLUGIN),
         "Unable to refresh service account access token.", e.getMessage(),
         ErrorType.UNKNOWN, true, e);
```

Comment:
better to use https://github.com/data-integrations/google-cloud/blob/c13becec11d9e5333bc85573ebc0ef287e9aae57/src/main/java/io/cdap/plugin/gcp/common/GCPErrorDetailsProviderUtil.java#L117 with externalDocUrl as https://cloud.google.com/compute/docs/troubleshooting/troubleshoot-metadata-server